### ISSUE TITLE
Check for equivocations

### ISSUE THREAD

**Issue by bkchr** [paritytech/cumulus#492]
Equivocation checking for Aura is currently disabled, because it only checks if a block producer has built multiple blocks on the same height. However, this is completely legal for a collator to build multiple blocks on the same height, as long as they are build on different relay chain blocks.
So, this will require some changes on the Substrate site to make the equivocation checking more flexible.

For the relay chain provided consensus we should do something similar.

**Comment by bkchr** [comment]
From the relay chain side we can not do that much against this, however a good parachain implementation should try to reduce the traffic with the relay chain.

**Comment by burdges** [comment]
Interesting.. It's legal for a collator to make multiple blocks at the same height, but we do not necessarily need to make it legal for them to make multiple blocks at the same slot, and some anti-MEV tricks benefit if this were punishable or something.  

Ideally, I've vaguely imagined a collator facing a relay chain fork would make one block and one PoV block, but then multiple different candidate receipts arise that share the same availability chunks from the PoV.  Yet, this is impossible since once the relay parents diverge the rest diverges too.

**Comment by rphmeier** [comment]
re: https://github.com/paritytech/polkadot/issues/3307

The use of PreVFs for the faster PoV-fetching usecase requires reasonable uniqueness per relay-parent, so punishment for equivocations should be important even for equivocations made on the same relay-parent.

**Comment by burdges** [comment]
Yes, it's problematic if parachain block producers could spam everyone asking for their entire chain history.  We could design anti-capture protocols that protect against this, which work assuming 2/3rd of the parachain is honest, but the simplest would like be parachain consensus.
